//
// PROJECT:         Aspia Remote Desktop
// FILE:            proto/file_transfer_session.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

import "file_transfer_session_message.proto";

package aspia.proto.file_transfer;

message HostToClient
{
    RequestStatus status         = 1;
    DriveList drive_list         = 2;
    FileList file_list           = 3;
    DirectorySize directory_size = 4;
    FilePacket file_packet       = 5;
}

message ClientToHost
{
    DriveListRequest drive_list_request             = 1;
    FileListRequest file_list_request               = 2;
    DirectorySizeRequest directory_size_request     = 3;
    CreateDirectoryRequest create_directory_request = 4;
    RenameRequest rename_request                    = 5;
    RemoveRequest remove_request                    = 6;
    FileDownloadRequest file_download_request       = 7;
    FileUploadRequest file_upload_request           = 8;
    FilePacketRequest file_packet_request           = 9;
    FilePacket file_packet                          = 10;
}
